### PR TITLE
scdoc 1.4.1 (new formula)

### PR DESCRIPTION
**Pull request by deathlyfrantic**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/scdoc.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  sha256 "2bc96daae0ddfb858305e550eb00a544a7a6255edd0bd169dcb2fc0c035eca92"
+
+  def install
+    system "make", "LDFLAGS=", "all"
+    bin.install "scdoc"
+    man1.install "scdoc.1"
+  end
+
+  test do
+    File.write "test.1.scd", "thisisatest(1)\n\n# TEST\n\nthis is a test"
```
Comment (DomT4) [inline Formula/scdoc.rb]: Use `(testpath/"test.1.scd").write`

File: Formula/scdoc.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class Scdoc < Formula
+  desc "Tool designed to make the process of writing man pages more friendly"
+  homepage "https://git.sr.ht/~sircmpwn/scdoc/"
+  url "https://git.sr.ht/~sircmpwn/scdoc/snapshot/scdoc-1.4.1.tar.xz"
+  sha256 "2bc96daae0ddfb858305e550eb00a544a7a6255edd0bd169dcb2fc0c035eca92"
+
+  def install
+    system "make", "LDFLAGS=", "all"
```
Comment (DomT4) [inline Formula/scdoc.rb]: What are we overriding passing a blank `LDFLAGS` here?

File: Formula/scdoc.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    bin.install "scdoc"
```
Comment (DomT4) [inline Formula/scdoc.rb]: Is there not a `make install` we can use?

File: Formula/scdoc.rb (hunk 2 above)
Comment (deathlyfrantic) [reply Formula/scdoc.rb]: By default this includes `-static` in the LDFLAGS which as best I can tell just does not work on macOS.

File: Formula/scdoc.rb (hunk 3 above)
Comment (deathlyfrantic) [reply Formula/scdoc.rb]: There is but it uses the `install` command and some flags that macOS' built-in `install` doesn't support. I tried doing some search/replace stuff on the Makefile but it grew to be a huge mess. `bin.install` and `man1.install` just seemed cleaner.

File: Formula/scdoc.rb (hunk 1 above)
Comment (deathlyfrantic) [reply Formula/scdoc.rb]: Fixed in an amended commit.

File: Formula/scdoc.rb (hunk 3 above)
Comment (DomT4) [reply Formula/scdoc.rb]: Oh I assume they're using GNU extensions. It's probably something fairly simple to fix; usually you can support the BSD `install` fairly easily, you just end up having to take a few more steps to reach the same destination as you would with GNU `install`. I'll take a look.

**Comment by DomT4** [comment]
I've emailed upstream this patch:
```patch
From aae2f67b2d4eb1bd1b35880fd23746df3390c1da Mon Sep 17 00:00:00 2001
Date: Tue, 14 Aug 2018 02:32:22 +0100
Subject: [PATCH] Makefile: be compatible with BSD install

---
 Makefile | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index fa1c804..0bc21f5 100644
--- a/Makefile
+++ b/Makefile
@@ -37,8 +37,10 @@ clean:
 	rm -rf $(OUTDIR) scdoc scdoc.1
 
 install: all
-	install -Dm755 scdoc $(BINDIR)/scdoc
-	install -Dm644 scdoc.1 $(MANDIR)/man1/scdoc.1
+	install -d $(BINDIR)
+	install -d $(MANDIR)/man1
+	install -m755 scdoc $(BINDIR)/scdoc
+	install -m644 scdoc.1 $(MANDIR)/man1/scdoc.1
 
 check: scdoc scdoc.1
 	@find test -perm -111 -exec '{}' \;
-- 
2.18.0
```

**Comment by fxcoudert** [comment]
Given no action on this, we'll welcome a pull request for a new version when upstream moves.

**Pull request closed without merging**